synthesizer/src/vm/verify.rs

```diff
@@ -95,6 +96,12 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
         // First, verify the fee.
         self.check_fee(transaction, rejected_id)?;
 
+        // If the transaction has already been verified (as indicated by the transaction cache), skip verification.
+        if already_verified {
```

Comment:
For deployments, I think the following similar checks cannot be cached / can grow stale:
- `self.transaction_store().contains_program_id(deployment.program_id())` (in synthesizer/src/vm/verify.rs)
- `self.contains_program(program_id)` (in synthesizer/process/src/verify_deployment)

For executions, I think the following cannot be cached / can grow stale:
- `self.block_store().contains_rejected_deployment_or_execution_id(&execution_id)` (in synthesizer/src/vm/verify.rs)
- `self.block_store().contains_state_root(&execution.global_state_root())` (in synthesizer/src/vm/verify.rs)

So I think we should pass the `already_verified` flag down into `check_{deployment, execution}_internal`